### PR TITLE
🐛(frontend) fix future course access link in dashboard

### PR DESCRIPTION
**Pull request by rlecellier**
## Purpose
In Learner dashboard when a course isn't yet open, we displayed a
"access course" link that was broken.
The LMS return a 500 for theses courses and the user arrived a blank
page.

There we decide to hide theses link and we use this change to upgrade
the period label of the enrollment.

### CODE REVIEW

File: src/frontend/js/widgets/Dashboard/components/DashboardItem/DashboardItemCourseEnrolling.tsx [hunk 1]
```diff
@@ -208,6 +227,28 @@ interface DashboardItemCourseEnrollingRunProps {
   product?: Product;
 }
 
+const getCourseRunPeriodMessage = (courseRun: CourseRun, selected: boolean = false) => {
```
Comment (jbpenrath) [inline src/frontend/js/widgets/Dashboard/components/DashboardItem/DashboardItemCourseEnrolling.tsx:230]: `selected` === `enrolled` ?

File: src/frontend/js/widgets/Dashboard/components/DashboardItem/DashboardItemCourseEnrolling.tsx [hunk 2]
```diff
@@ -41,15 +41,31 @@ const messages = defineMessages({
...
+    defaultMessage: 'This session will run from {startDate} to {endDate}',
+  },
+  onGoingEnrolledRunPeriod: {
+    id: 'components.DashboardItemEnrollment.onGoingEnrolledRunPeriod',
+    description: 'Text to display the period of an ongoing course run that user is enrolled to.',
+    defaultMessage: "You are enrolled for this session. It's open from {startDate} to {endDate}",
+  },
+  futureEnrolledRunPeriod: {
+    id: 'components.DashboardItemEnrollment.futureEnrolledRunPeriod',
+    description: 'Text to display the period of a future course run that user is enrolled to.',
```
Comment (jbpenrath) [inline src/frontend/js/widgets/Dashboard/components/DashboardItem/DashboardItemCourseEnrolling.tsx:61]:
```suggestion
    description: 'Text to display when a course run is not yet opened and the user is enrolled to.',
```

File: src/frontend/js/widgets/Dashboard/components/DashboardItem/DashboardItemCourseEnrolling.tsx [hunk 3]
```diff
@@ -41,15 +41,31 @@ const messages = defineMessages({
...
+    defaultMessage:
+      'You are enrolled for this session. It will be open from {startDate} to {endDate}',
```
Comment (jbpenrath) [inline src/frontend/js/widgets/Dashboard/components/DashboardItem/DashboardItemCourseEnrolling.tsx:63]: Ditto Somewhere else, we are displaying in how many times the course will starts. IMO, it's the more relevant information.

Comment (rlecellier) [reply src/frontend/js/widgets/Dashboard/components/DashboardItem/DashboardItemCourseEnrolling.tsx:63]: I don't know, if i'm a client i like to have a date to log in my calendar. Not to compute "in 14 days", that's mean Wednesday in two week.

**Review by jbpenrath — APPROVED** [review]

**Pull request merged**